Snowflake.Data/Core/SFStatement.cs

```diff
@@ -182,6 +185,7 @@ private SFBaseResultSet BuildResultSet(QueryExecResponse response, CancellationT
         {
             if (response.success)

```

Comment:
QueryId is also for failure responses and imho it makes sense to provide it in any case if only server has supplied it in its response. 
That's why I would ask for this change @sfc-gh-ext-simba-hx but let's wait for @sfc-gh-igarish to confirm.

Reply:
done